### ISSUE TITLE
Support for Apple GPUs (Apple M2 Max)

### ISSUE THREAD

**Issue by ncbss** [MIC-DKFZ/HD-BET#61]
Hi there,
Thanks for this awesome software. 

I saw a couple of comments about making HD-BET compatible with Apple GPUs but it was not clear whether this is the case. Can you guys confirm?

Thanks

**Comment by mrunibe** [comment]
I have successfully used HD-BET on Apple Silicon using MPS backend. The performance is good, much faster than running on CPU.

However, some minor modifications are required, see corresponding PR:

https://github.com/MIC-DKFZ/HD-BET/pull/55

@FabianIsensee any plans to merge this PR?

**Comment by FabianIsensee** [comment]
We had a new release on Friday which now bases hd-bet on [nnU-Net](https://github.com/MIC-DKFZ/nnUNet). With that we should have support for mps. I could not test this though as I don't have access to a mac

**Comment by ncbss** [comment]
Hi there, thanks for this! I was able to the test with the modifications suggested in your previous posts (#55). However, I ran into this issue with the newer version of HD-BET ```RuntimeError: ConvTranspose 3D is not supported on MPS```. Any suggestions?

Thanks again for the help!

**Comment by mrunibe** [comment]
> Hi there, thanks for this! I was able to the test with the modifications suggested in your previous posts ([#55](https://github.com/MIC-DKFZ/HD-BET/pull/55)). However, I ran into this issue with the newer version of HD-BET `RuntimeError: ConvTranspose 3D is not supported on MPS`. Any suggestions?
> 
> Thanks again for the help!

If you set:

`export PYTORCH_ENABLE_MPS_FALLBACK=1`

It should work (falling back to CPU for non implemented ops).

**Comment by FabianIsensee** [comment]
Yep Apple has not implemented some important functionality. Sorry for overlooking that!
